packages/editor/src/extensions/code-block/highlighter.ts

```diff
@@ -307,6 +314,11 @@ function updateSelection(
       isDocChanged ? toCodeLines(node.textContent, pos) : undefined
     );
     attributes.caretPosition = position;
+    console.log({
+      attrs: node.attrs,
+      defaultLanguage: defaultLanguage()
+    });
+    attributes.language = node.attrs.language ?? defaultLanguage();
```

Comment:
This works but I think it'll cause issues with old code blocks (added before this change) as they'll get their language changed to the cached language as soon as the user makes a selection or a change in the document.

We should find a way to only run this once when the code block is added in the document. I think we can compare `oldState` and `newState` to find that out.